backend/app/domains/habit_logs/schemas.py

```
@@ -0,0 +1,48 @@
+from app.domains.shared.schemas import BaseSchema
+
+
+class HabitLogBase(BaseSchema):
+    """Base habit log schema."""
+    local_date: date = Field(..., description="Date in user's timezone")
+    meta: Optional[dict] = Field(default=None, description="Additional metadata")
+
+
+class HabitLogCreate(HabitLogBase):
+    """Schema for creating a habit log."""
+    idempotency_key: Optional[str] = Field(default=None, max_length=255)
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)